packages/@aws-cdk/core/lib/deps.ts

```diff
@@ -13,18 +13,55 @@ type Element = CfnResource | Stack;
  * - Try to find the deepest common stack between the two elements
  * - If there isn't a common stack, it means the elements belong to two
  *   disjoined stack-trees and therefore we apply the dependency at the
- *   assembly/app level between the two topl-level stacks.
+ *   assembly/app level between the two top-level stacks.
  * - If we did find a common stack, we apply the dependency as a CloudFormation
  *   "DependsOn" between the resources that "represent" our source and target
  *   either directly or through the AWS::CloudFormation::Stack resources that
  *   "lead" to them.
  *
- * @param source The source resource/stack (the depedent)
+ * @param source The source resource/stack (the dependent)
  * @param target The target resource/stack (the dependency)
  * @param reason Optional resource to associate with the dependency for
  * diagnostics
  */
-export function addDependency<T extends Element>(source: T, target: T, reason?: string) {
+export function addDependency<T extends Element>(source: T, target: T, reason: string='dependency added using stack.addDependency()') {
```

Comment:
Looks like the changes in this file don't have corresponding unit tests. Please add those.